gugel-project-2/App.js

```diff
@@ -0,0 +1,90 @@
+// In App.js in a new project
+import * as React from 'react';
+import { View, FlatList, ScrollView, Text, Button, SectionList, StyleSheet, TextInput, TouchableWithoutFeedback } from 'react-native';
+import {List, ListItem, SearchBar } from "react-native-elements";
+import { NavigationContainer, useNavigation, withNavigation } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+
+let movieList = [{title: 'Movie A', date: 1997, key: '1'}, {title: 'Movie B', date: 2000, key: '2'}, {title: 'Movie C', date: 1942, key:'3'}]
+const styles = StyleSheet.create({
+  title: {
+    fontSize: 48
+  },
+})  
+let ogMovieList = movieList
+let itemData = ''
+let textData = ''
+let newData = []
+
+let searchFilterFunction = (newData) => {    
+  setState({newData})
+}
+
+// this does nothing, but that's ok. it just needs to run to update state.
+let updateSearch = (search) => {}
+
+function HomeScreen({ navigation }) {
+  const [ search, updateSearch] = React.useState('');
+  let [ movieData, searchFilterFunction] = React.useState(movieList);
```

Comment:
I don't think we really need the movie data in state, we can just store the filtered data in a variable
```suggestion
  const movieData = filteredMovieData(search);
```
then you'd want to move your filter from the search bar to a new `filteredMovieData` function